### DISCUSSION TITLE
How to use osqp in Simulink with matlab version 2015b?

### DISCUSSION THREAD

**Discussion by DRETQ** [osqp/osqp#656] [Matlab/Simulink Interface]
I want to call osqp on Matlab/Simulink version 2015b, what is the best way to get it through compilation and generate C code?

**Comment by imciner2** [comment]
Hi @DRETQ, it has been a while since I used OSQP with Simulink, and especially one that is that old. I had used it on Simulink 2017 I believe, and that was based on this branch I kept in a personal repository: https://github.com/imciner2/osqp-matlab/tree/im/simulink_cg.

The main problem is the Simulink codegen system is highly dependent on the internals of Simulink, and so I am not entirely sure how portable this is between versions. I am planning to update this to be better supported for newer Matlab/Simulink versions, but can I ask what the reason for using 2015b is?

**Comment by DRETQ** [comment]
@imciner2 Thanks for your reply, the reason is that I need to design quadratic programming algorithms for a model written in matlab/simulink 2015b

**Comment by DRETQ** [comment]
@imciner2 Sorry to bother you again, but did you use the S-Function Builder module to call the existing c-code and implement it in simulink?

**Comment by DRETQ** [comment]
@imciner2  How did you package osqp into a library in the lower version matlab/simulink 2017, can it be done in the lower version 2015b？

**Reply by ywxxxxxx** [reply]
hello，have you solved this problem？ I meet the same problem as yours. Hope that you have finished that and give some help for me.Thank you.

**Reply by DRETQ** [reply]
I haven't solved this problem yet, the suggestion is to use the code generation on the osqp official website, write a c mex s function and then call it in simulink

**Reply by ywxxxxxx** [reply]
It seems that simulink's code generate C code can't work normally, if take this way...

**Reply by DRETQ** [reply]
Refer to this URL https://osqp.org/docs/codegen/matlab.html
